### PR TITLE
Close dialogs on click outside the dialog

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**

### CODE REVIEW

File: services/frontend-service/src/ui/components/dialog/ConfirmationDialog.tsx [hunk 1]
```diff
@@ -73,7 +73,19 @@ export const PlainDialog: React.FC<PlainDialogProps> = (props) => {
                 onClose();
             }
         });
-    }, [onClose]);
+        document.addEventListener('click', (event) => {
+            if (open) {
+                if (event.target instanceof HTMLElement) {
+                    // eslint-disable-next-line no-type-assertion/no-type-assertion
+                    const target = event.target as HTMLElement;
```
Comment (jaypontes) [inline services/frontend-service/src/ui/components/dialog/ConfirmationDialog.tsx:80]: Is there no way to get around the `as HTMLElement`? :(

Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/components/dialog/ConfirmationDialog.tsx:80]: I am not aware of one.

`event.target`s type is very generic.

Comment (jaypontes) [reply services/frontend-service/src/ui/components/dialog/ConfirmationDialog.tsx:80]: Having the condition in the line 78 doesn't remove the need for the cast?

**Review by jaypontes — APPROVED** [review]

File: services/frontend-service/src/ui/components/dialog/ConfirmationDialog.tsx (hunk 1 above)
Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/components/dialog/ConfirmationDialog.tsx]: You are right, it works :smiley: 
I simplified it

**Pull request merged**